_posts/2023-11-30-prevent-broker-scale-down-if-containing-paritition-replicas.md

```diff
@@ -0,0 +1,179 @@
+If partition replicas are found on the broker then the cluster operations are blocked and reconciliation fails until you revert the Kafka resource.
+This is enabled by default with Strimzi 0.38.
+
+However, there may be scenarios where you want to bypass this blocking mechanism.
+For example, disabling the check might be necessary when new topics are being constantly created.
+This situation can indefinitely block cluster operations, even when brokers are nearly empty.
+
+You can bypass the blocking mechanism by annotating the `Kafka` resource for the Kafka cluster by setting the `strimzi.io/skip-broker-scaledown-check` annotation to `true`:
+```shell
+kubectl annotate Kafka my-cluster strimzi.io/skip-broker-scaledown-check="true"
+```
+
+## Setting up the environment
+
+Let's set up a cluster to work through an example demonstrating this feature.
+
+To get the Kafka cluster up and running, we will first have to install the Strimzi Cluster Operator and then deploy the `Kafka` resource.
+You can refer to the [Stimzi Quickstart Guide](https://strimzi.io/docs/operators/latest/quickstart.html) to install the operator.
+
+You can install the Cluster Operator with any installation method you prefer.
+
+Then we will deploy the Kafka cluster with Cruise Control enabled.
+
+Example Kafka resource with Cruise Control enabled:
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    version: 3.6.0
+    replicas: 4
+    listeners:
+      - name: plain
+        port: 9092
+        type: internal
+        tls: false
+      - name: tls
+        port: 9093
+        type: internal
+        tls: true
+    config:
+      offsets.topic.replication.factor: 3
+      transaction.state.log.replication.factor: 3
+      transaction.state.log.min.isr: 2
+      default.replication.factor: 3
+      min.insync.replicas: 2
+      inter.broker.protocol.version: "3.6"
+    storage:
+      type: ephemeral
+  zookeeper:
+    replicas: 3
+    storage:
+      type: ephemeral
+  entityOperator:
+    topicOperator: {}
+    userOperator: {}
+  cruiseControl: {}
+```
+
+Once the cluster is up, we can create some topics such that we have some partition replicas assigned to the brokers.
+This is a topic configuration example:
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaTopic
+metadata:
+  name: my-topic
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  partitions: 3
+  replicas: 3
+  config:
+    retention.ms: 7200000
+    segment.bytes: 1073741824
+```
+
+You can now check how topic partition replicas are allocated across the brokers using the command:
+```shell
+$ kubectl run -n myproject client -itq --rm --restart="Never" --image="quay.io/strimzi/kafka:latest-kafka-3.6.0" -- \
+sh -c "/opt/kafka/bin/kafka-topics.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --describe --topic my-topic; exit 0";
```

Comment:
Wouldn't this work as well and be easier to understand for users? You should also use a release image I guess.

```suggestion
$ kubectl run -n myproject client -itq --rm --restart="Never" --image="quay.io/strimzi/kafka:0.38.0-kafka-3.6.0" -- /opt/kafka/bin/kafka-topics.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --describe --topic my-topic
```